### ISSUE TITLE
Causer in CLI Commands.

### ISSUE THREAD

**Issue by nospam4u** [spatie/laravel-activitylog#649]
So I have been integrating ActivityLog into all our apps and it has worked perfectly for us.  I do have a question about how I can set a causer for console commands.  We have one environment where a great deal of the management orchestration takes place at the CLI, and I would like to track the Linux username of the person executing those commands.  Since the Linux user doesn't exist as a Laravel user, I can't just use ->causedBy().  Currently I am just adding it to the description, but would like to be able to query these in their own column for reporting.

Are there any secrets I am missing, or is adding to to description or properties the best available method?

Thanks!

**Comment by Gummibeer** [comment]
Hey,
you have multiple ways to achieve this.

You could create another user model/table and relate the activity to it. The causer is also morphable, so no problem.

You could also use the json column custom properties and add it there, via JSON queries (MySQL 5.7, postgres ... support it).

Or you add a new custom column with your own migration & model and use the `tap()` method or even an observer to fill it.

**Comment by Gummibeer** [comment]
I will close this issue even if v4 isn't released yet. But the task itself is done and I want to check which tasks are really open. Please keep an eye on #787

**Comment by freekmurze** [comment]
Addressed in v5 (#1452) via `CauserResolver::withCauser()`. You can now set the causer for CLI commands:

```php
CauserResolver::withCauser($systemUser, function () {
    // activities logged with $systemUser as causer
});
```
